lib/core_extensions.rb

```diff
@@ -39,9 +39,17 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
-          record.errors.add :base, _("%{record} is used by %{what}") % { :record => record, :what => what }
+        record.association(klass.to_sym).association_scope.each do |what|
+          error_message = _("%{record} is used by %{what}.")
```

Comment:
Nitpick, remove the full stop from the end as model errors tend to be concatenated together with commas by Rails.